tests/test_examples.py

```diff
@@ -1026,3 +1026,127 @@ async def test_38_shutdown_action():
     assert event["action"] == "shutdown", "1"
     event = event_log.get_nowait()
     assert event["type"] == "Shutdown", "2"
+
+
+@pytest.mark.asyncio
+@pytest.mark.skipif(
+    os.environ.get("EDA_RULES_ENGINE", "durable_rules") == "durable_rules",
+    reason="durable rules does not support time based rules",
+)
+async def test_40_coalesce_events_single_host():
```

Comment:
How the test verifies that the time window is honored?

Reply:
@Alex-Izquierdo Most of the time validation would have to be tested in integration tests, we are implementing the unit tests in drools_jpy by advancing the clock. Didn't want to repeat those same tests here.
https://github.com/ansible/drools_jpy/blob/1ce4fca15553cd9b77e5226c3d700a1daca1d5aa/tests/test_ruleset.py#L508
Since we have to advance the clock and we are using 2 different languages (python and java) we can't use the time travel python packages like **freeze_gun** to move the time forward. The Java layer allows us to create a pseudo clock on the Java side and we can advance it like here to expedite the tests.
https://github.com/ansible/drools_jpy/blob/1ce4fca15553cd9b77e5226c3d700a1daca1d5aa/tests/test_ruleset.py#L549

Reply:
Ok, It was only curiosity if I missed something. Anyway, I will test it as well in the future e2e tests, so everything is fine. :)